tests/core/evm/vm/test_message.py

```diff
@@ -0,0 +1,32 @@
+import pytest
+import evm.vm.message as m
+
+message_object = m.Message(
+    1,                      #gas
+    1,                      #gas_price
+    b'',                    #to
+    '11111111111111111111', #sender
+    1,                      #value
+    '',                     #data
+    'asdf',                 #code
+    '22222222222222222222'  #origin
+    )                    
+
+def test_initilizes_properly():
+    # all __init__ parameters are properly validated.
+    assert message_object.gas == 1
+    assert message_object.gas_price == 1
+    assert message_object.to == b''
+    assert message_object.sender == '11111111111111111111'
+    assert message_object.value == 1
+    assert message_object.data == ''
+    assert message_object.code == 'asdf'
+    assert message_object.origin == '22222222222222222222'
+
+def test_is_origin():
```

Comment:
This tests the case where it is not the origin, but not the positive case where it is the origin.

There are two ways you could change this.

1. rename this test to `test_message_is_not_origin_when_sender_and_origin_address_are_different`, and then write another new test called `test_message_is_origin_when_sender_and_origin_are_same`.  Note the very verbose test names make it extremely clear what is being tested.
2. Use the `pytest.mark.parametrize` marker to write a generic tests that runs multiple times with different parameters.  See below.

```
@pytest.mark.parametrize(
    "origin,sender,should_be_origin",
    (
        ("0x01", "0x02", False),  # tests that when they are diffrent, it isn't the origin message.
        ("0x01", "0x01", True),  # tests that when they are the same, it is the origin message.
    )
)
def test_computed_is_origin_property(origin, sender, should_be_origin):
    message = _create_message(sender=sender, origin=origin)
    assert message.is_origin is should_be_origin
```

Note that you need to use different values than `"0x01"` and `"0x02"` as those are just meant to be illustrative.